### PR TITLE
Update Node.js LTS support policy

### PR DESCRIPTION
**Pull request by alexandrebouthinon**
## What does this PR do ?

Update our processes to match the newly adopted Node.js LTS version support decisions.

Kuzzle will follow the same LTS policy as Node.js and thus always support 3 version at the same time.

The 12 April 2022, Kuzzle will support only the following versions:
 - Node.js 14
 - Node.js 16
 - Node.js 18

- [x] Update Github Actions Workflows
- [x] Update Docker base images
- [x] Build and deploy Docker image for both amd64 and arm64
- [x] Rework CI pipeline to make it one file using condition

DISCLAIMER: This is normal all the deployment jobs (Docker, Doc) can be seen in the Pull request workflow. This a weird way Github display them even if the running condition is `false`

### How should this be manually tested?

Check CI/CI pipelines

### CODE REVIEW

File: .github/workflows/workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        node-version: ${{ fromJson(needs.prepare-matrix.outputs.matrix).node-version }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./.github/actions/install-packages
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - uses: ./.github/actions/cache-node-modules
+        with:
+          NODE_VERSION: ${{ matrix.node-version }}
```
Comment (Aschen) [inline .github/workflows/workflow.yaml:77]: Is it normal to have uppercase and lowercase?

File: .github/workflows/workflow.yaml [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        with:
+          KUZZLE_IMAGE: ${{ matrix.kuzzle-image }}
+
+
+  functional-tests:
+    name: Functional Tests - ${{ matrix.test_set }} - Node.js v${{ matrix.node-version }}
+    needs: [unit-tests, prepare-matrix]
+    strategy:
+      matrix:
+        test_set: [http, websocket, "legacy:mqtt", "legacy:http", "legacy:websocket"]
```
Comment (Aschen) [inline .github/workflows/workflow.yaml:125]: Mix of strings with `"` and without

Comment (alexandrebouthinon) [reply .github/workflows/workflow.yaml:137]: This is normal, Github Actions try to interpret the `:` as a YAML property, I don't know really why 😓

File: .github/workflows/workflow.yaml (hunk 1 above)
Comment (alexandrebouthinon) [reply .github/workflows/workflow.yaml:85]: We could use `node-version` in our custom cache action but this is not the main goal of this PR. I can update it if it is a big concern for you 🙂

Comment (alexandrebouthinon) [reply .github/workflows/workflow.yaml]: Updated 😉

**Review by Aschen — APPROVED** [review]

**Review by Shiranuit — APPROVED** [review]

File: docker/images/kuzzle/Dockerfile [hunk 3]
```diff
@@ -1,20 +1,18 @@
 ################################################################################
 # Production build image
 ################################################################################
-FROM node:12.18.1-stretch-slim as builder
+FROM node:12-bullseye-slim as builder
```
Comment (Aschen) [inline docker/images/kuzzle/Dockerfile:4]: Shouldn't we build those images with newer version of node?

**Pull request merged**